### ISSUE TITLE
Suggestion: Allow RemoveTransactions to resync if previously called without resync

### ISSUE THREAD

**Issue by sondreb** [stratisproject/StratisBitcoinFullNode#2655]
The RemoveTransactions ("remove-transactions") method on the WalletController takes a model (from query string) that supports "all" and "reSync". If "all" is supplied, without "reSync", then all of the transaction history is cleared. Calling it again with reSync as true, does not start sync, as it require there to be any resulting deleted transactions, for a resync to be called.

```C#
// If the user chose to resync the wallet after removing transactions.
if (result.Any() && request.ReSync)
```

If calling first with "all" as true, then second call with not have any deleted transactions returned, and hence won't resync.

Here is a suggested change, it will have to perform a check if there is any result, then previous logic is used and creation time of the last TRX is used. Else, we'll get the creation time from the wallet.

```C#
// If the user chose to resync the wallet after removing transactions.
if (request.ReSync)
{
    // Update the wallet and save it to the file system.
    Wallet wallet = this.walletManager.GetWallet(request.WalletName);

    DateTimeOffset earliestDate;

    if (result.Any())
    {
        // From the list of removed transactions, check which one is the oldest and retrieve the block right before that time.
        earliestDate = result.Min(r => r.creationTime);
    }
    else {
        earliestDate = wallet.CreationTime;
    }
```

SyncFromDate could be an alternative, but it require a date to be specified. And I think it applied to all wallets, as there are no parameter to specify wallet, which there is on the RemoveTransactions method.

Is this something that would be interesting to support, then I could make an PR.

**Comment by bokobza** [comment]
Yeah, the resync is there to sync IF something has been removed from the wallet.
In order to sync independently from transactions removal, we should use `api/wallet/sync` or `api/wallet/syncfromdate`.
We could potentially allow resyncing even if nothing has been removed. When do you think such a scenario might happen?

Your suggested change will sync from CreationTime if you try to remove even 1 transaction that wasn't there so we'd need to be a bit more specific.

**Comment by sondreb** [comment]
Scenario I came across when I discovered this, was that I called the RemoveTransactions without resync, and called it again resync, and it wouldn't sync. So I debugged and saw it required finding something to delete, to perform a resync.

The problem has been that 3-8% of transactions that users send ends up in a "stale" state, so I needed an option to resync history. Now I always send both "all" and "resync", so not really an issue.

I also started working on a re-broadcast option, and I was thinking I could also have a "remove transaction" action, which calls RemoveTransactions with the ids so it only remove that single TRX.

wallet/sync and syncfromdate are alternatives, but as I mentioned, they require that user unlocks their wallet before calling those methods, while the RemoveTransactions can be ran on any wallet on the node.
